### PR TITLE
Add mg

### PR DESCRIPTION
**Pull request by gerardhros**
The following functions are added

1. crop classification (sensitivity to P, K and S supply)
2. soil organic matter balance (simple one)
3. risk for soil compaction  (following risk map Van den Akker, 2013)
4. index and evaluation of Mg-availability
5. index and evaluation of S-availability (evaluation function need still to be parameterized)
6. index and evaluation of winderodibility
7. index and evaluation of soil strategic management (following Label Duurzaam Bodembeheer)

### CODE REVIEW

File: R/sulpher.R [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+  setkey(soils.obic, soiltype)
+  
+  # Check input
+  arg.length <- max(length(D_SLV), length(B_LU_BRP), length(B_BT_AK), length(B_LG_CBS))
+  checkmate::assert_numeric(D_SLV, lower = 0, upper = 150, any.missing = FALSE, len = arg.length)
+  checkmate::assert_numeric(B_LU_BRP, any.missing = FALSE, min.len = 1, len = arg.length)
+  checkmate::assert_subset(B_LU_BRP, choices = unique(crops.obic$crop_code), empty.ok = FALSE)
+  checkmate::assert_character(B_BT_AK, any.missing = FALSE, min.len = 1, len = arg.length)
+  checkmate::assert_subset(B_BT_AK, choices = unique(soils.obic$soiltype), empty.ok = FALSE)
+  checkmate::assert_character(B_LG_CBS, any.missing = FALSE, min.len = 1, len = arg.length)
```
Comment (SvenVw) [inline R/sulpher.R]: Add maybe assert_subset() for B_LG_CBS

File: tests/testthat/test-compaction.R [hunk 2]
```diff
@@ -0,0 +1,17 @@
+require(data.table);require(testthat)
```
Comment (SvenVw) [inline tests/testthat/test-compaction.R]: line should be removed

File: tests/testthat/test-crop_classification.R [hunk 3]
```diff
@@ -0,0 +1,43 @@
+require(data.table);require(testthat)
```
Comment (SvenVw) [inline tests/testthat/test-crop_classification.R]: line should be removed

File: tests/testthat/test-management.R [hunk 4]
```diff
@@ -0,0 +1,94 @@
+require(data.table);require(testthat)
```
Comment (SvenVw) [inline tests/testthat/test-management.R]: Line should be removed

File: tests/testthat/test-winderodibility.R [hunk 5]
```diff
@@ -0,0 +1,44 @@
+require(data.table);require(testthat)
```
Comment (SvenVw) [inline tests/testthat/test-winderodibility.R]: Line should be removed

File: R/crop_rotation.R [hunk 6]
```diff
@@ -21,7 +21,9 @@ calc_rotation_fraction <- function(ID, B_LU_BRP, crop) {
   checkmate::assert_numeric(ID, len = arg.length)
   checkmate::assert_numeric(B_LU_BRP, any.missing = FALSE, len = arg.length)
   checkmate::assert_subset(B_LU_BRP, choices = unique(crops.obic$crop_code), empty.ok = FALSE)
-  checkmate::assert_choice(crop, choices = c("starch", "potato", "sugarbeet", "grass", "mais", "other"))
```
[the remark this answers is not in the document]
Comment (gerardhros) [reply R/crop_rotation.R]: solved by reclassification rustgewassen in category "other" for the case that the function is not used for classification 'rustgewas'

File: R/crop_classification.R [hunk 7]
```diff
@@ -0,0 +1,156 @@
+#' Determine classification rules for crops used to prepare crops.obic
```
[the remark this answers is not in the document]
Comment (gerardhros) [reply R/crop_classification.R]: Agronomic crop sensitivity classes (response to fertilizers) not only depends on crop type but also on soil type. So, its not only a crop property. Therefore, i would not add this to the crops.obic table.

**Pull request merged**